pkg/cmd/server/origin/master.go

```diff
@@ -477,7 +477,30 @@ func (c *MasterConfig) ensureMasterAuthorizationNamespace() {
 	// ensure that master namespace actually exists
 	namespace, err := c.KubeClient().Namespaces().Get(c.Options.PolicyConfig.MasterAuthorizationNamespace)
 	if err != nil {
-		namespace = &kapi.Namespace{ObjectMeta: kapi.ObjectMeta{Name: c.Options.PolicyConfig.MasterAuthorizationNamespace}}
+		namespace = &kapi.Namespace{
+			ObjectMeta: kapi.ObjectMeta{Name: c.Options.PolicyConfig.MasterAuthorizationNamespace},
+			Spec: kapi.NamespaceSpec{
+				Finalizers: []kapi.FinalizerName{kapi.FinalizerKubernetes, projectapi.FinalizerProject},
+			},
+		}
+		kapi.FillObjectMetaSystemFields(api.NewContext(), &namespace.ObjectMeta)
+		_, err = c.KubeClient().Namespaces().Create(namespace)
```

Comment:
the kapi.FinalizerKubernetes is, but the FinalizerProject is not (unless you go to the project endpoint, which this is not)